File: day6/second.py
with open("input.txt") as file:
    content = file.read()
    char_set = set()
    char_count = 0
    for i in range(len(content) - 14):
        char_count += 1
        chars = content[i:i+14]
        if len(chars) == len(set(chars)):
            print(char_count + 13)
            break

File: day10/second.rb
puts "hello"
special_cycles = [20, 60, 100, 140, 180, 220]
result = 0
x = 1
screen = "." * 240
File.open("input.txt") do |file|
    content = file.read()
    lines = content.split("\n")
    cycle = 1
    lines.each do |line|
      if line == "noop"
        if (x-1..x+1).include?((cycle - 1) % 40)
          screen[cycle - 1] = "#"
        end
        cycle += 1
      else
        round = line.split.last.to_i
        if (x-1..x+1).include?((cycle - 1) % 40)
          screen[cycle - 1] = "#"
        end
        cycle += 1
        if (x-1..x+1).include?((cycle - 1) % 40)
          screen[cycle - 1] = "#"
        end
        cycle += 1
        x += round
      end
    end
    puts screen.chars.each_slice(40).map(&:join)
    puts screen.size
end
